### ISSUE TITLE
Dictionary inherits

### ISSUE THREAD

**Issue by georgeyanev** [travelping/ergw_aaa#214]
In `diameter_3gpp_ts32_299_ro.dia` you have `@inherits diameter_rfc7155_nasreq`
And in `diameter_3gpp_ts29_212.dia` which is also inherited by `diameter_3gpp_ts32_299_ro.dia` you have
`@inherits diameter_rfc4005_nasreq`.

Wouldn't these two versions of the nasreq application participating in the same final dictionary cause a potential conflict?

**Comment by vkatsuba** [comment]
The `@inherits directive` in Diameter dictionary files is used to indicate that the new dictionary file inherits definitions from another dictionary file. In the case you've described, there seems to be a potential conflict if both `diameter_3gpp_ts32_299_ro.dia` and `diameter_3gpp_ts29_212.dia` inherit different versions of the same application.

This scenario may lead to conflicts if there are conflicting definitions or changes in the inherited dictionaries. However, the exact impact depends on the specifics of the Erlang `Diameter` application and how it is implemented in the `Diameter` stack you are using.

Also, please notes that `diameter_3gpp_ts29_212` is used at [ergw_aaa_gx.erl](https://github.com/travelping/ergw_aaa/blob/cbfdf1648ff4e64b59cc1c3d2e7794d70fef81cc/src/ergw_aaa_gx.erl#L44) and `diameter_3gpp_ts32_299_ro` at [ergw_aaa_ro.erl](https://github.com/travelping/ergw_aaa/blob/cbfdf1648ff4e64b59cc1c3d2e7794d70fef81cc/src/ergw_aaa_ro.erl#L43) - all behaviors are covered by common tests as well and similar issue which was described has not appeared before.

**Comment by georgeyanev** [comment]
Yes, I think the condition you describe here is met because both `diameter_3gpp_ts32_299_ro.dia` and `diameter_3gpp_ts29_212.dia` inherit different versions of the nasreq application. And as I found out both nasreq versions are quite different.
Should these dictionaries be fixed in a way that for example `diameter_3gpp_ts32_299_ro.dia` does not happen to inherit more than one version of nasreq transitively ?(or any other app which happen to have multiple versions)? 
Actually I'm playing with `go-diameter` stack and since your dictionaries seemed quite complete I was trying to convert some of them to the xml format required by go-diameter.

**Comment by RoadRunnr** [comment]
inheritance in the Erlang Diameter is not true OOP style inheritance. It imports only the AVPs and later AVP declarations will override one that where encountered earlier.

When you are looking for XML Diameter dictionaries, then checkout wireshark. All dictionaries in there are in XML.
